File: c++/18111.cpp
#include <climits>
#include <iostream>

using namespace std;

int n, m, b;
int arr[501][501];
int answerTime = INT_MAX;
int answerHeight = 0;

int main() {
  freopen("example.txt", "r", stdin);

  cin >> n >> m >> b;

  for (int i = 0; i < n; i++) {
    for (int j = 0; j < m; j++) {
      cin >> arr[i][j];
    }
  }

  for (int h = 0; h <= 256; h++) {
    int createCnt = 0;
    int removeCnt = 0;

    for (int i = 0; i < n; i++) {
      for (int j = 0; j < m; j++) {
        int gap = arr[i][j] - h;
        if (gap < 0)
          createCnt -= gap;
        else
          removeCnt += gap;
      }
    }

    if (createCnt - removeCnt <= b) {
      int time = 2 * removeCnt + createCnt;
      if (time <= answerTime) {
        answerHeight = h;
        answerTime = time;
      }
    }
  }

  cout << answerTime << ' ' << answerHeight;

  return 0;
}

File: js/18111.js
const fs = require('fs');

const inputs = fs.readFileSync('/dev/stdin').toString().trim().split('\n');

const [n, m, b] = inputs[0].split(' ').map(Number);
const arr = [];
for (let i = 1; i <= n; i++) {
  arr.push(inputs[i].split(' ').map(Number));
}

let answerTime = Infinity;
let answerHeight = -1;

for (let h = 0; h <= 256; h++) {
  let createCnt = 0;
  let removeCnt = 0;

  for (let i = 0; i < n; i++) {
    for (let j = 0; j < m; j++) {
      let gap = arr[i][j] - h;
      if (gap < 0) createCnt -= gap;
      else removeCnt += gap;
    }
  }
  if (createCnt - removeCnt <= b) {
    let time = 2 * removeCnt + createCnt;
    if (answerTime >= time) {
      answerTime = time;
      answerHeight = h;
    }
  }
}

console.log(answerTime, answerHeight);
